### DISCUSSION TITLE
Save position for long file playlists

### DISCUSSION THREAD

**Discussion by loop333** [jcorporation/myMPD#1319] [Ideas]
I have several long playlist files (label packs) and it would be great to have opportunity to remember current play position in these playlists. So switching from playlist playback to playing other album or playlist and then back continues playback at saved position. This could be done by using scripts and triggers by including current playlist filename to mympd_state.

**Comment by jcorporation** [comment]
MPD 0.24 brings stickers for playlists. This will be the best place to remember the playlist position. 

Support for this will be included in myMPD v18.

myMPD saves the last played time of each songs in stickers. The last played song of a playlist is already included in the playlist content response, but not displayed in the gui.

**Reply by loop333** [reply]
quick temp solution:
[github](https://github.com/loop333/mympd_playlist_position)

**Comment by jcorporation** [comment]
I created an issue for this: https://github.com/jcorporation/myMPD/issues/1337

**Reply by loop333** [reply]
You’re right. Including playlist uri in MPD status is the best solution

**Reply by jcorporation** [reply]
Including the last loaded playlist does not solve our problem completely. As the user can modify the playlist we must always check if the last played song is really in the playlist and this could be an operation with much overhead.

I implemented this feature like commented in the above issue. Is this good enough for you?

**Reply by loop333** [reply]
My “use case” very simple:
1. I create m3u file with albums of some artist or label (by means of beets)
2. I listen to this playlist for several days
3. I switch to listening to new downloaded albums or my favorite old albums
4. I want to continue listening to m3u playlist (p.1) at beginning of last played track
5. Repeat p.3 and p.4
I don’t want to modify playlist till I end to listen to it )
So for me it’s good enough to have m3u playlist file name/url in mympd or mpd API )
